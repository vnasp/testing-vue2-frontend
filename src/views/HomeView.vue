<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <PostsView />
  </div>
</template>

<script>
import PostsView from './PostsView.vue';

// @ is an alias to /src

export default {
  name: 'HomeView',
  components: {
    PostsView
}
}
</script>
