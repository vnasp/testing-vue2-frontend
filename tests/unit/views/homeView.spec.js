import { mount } from '@vue/test-utils'
import HomeView from '@/views/HomeView.vue'

describe('HomeView', () => { 
    test('Crea / valida un snapshot de la vista HomeView', async () => {
      //Selección el Sujeto de pruebas
      const wrapper = mount(HomeView)

      //Aserción
      expect(wrapper.html()).toMatchSnapshot()
    })  
 })