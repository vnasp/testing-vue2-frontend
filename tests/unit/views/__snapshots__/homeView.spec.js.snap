// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Crea / valida un snapshot de la vista HomeView 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <div class="posts">
    <h1>Vista de Posts</h1>
    <ul>
      <li>Post 1</li>
      <li>Post 2</li>
      <li>Post 3</li>
      <li>Post 4</li>
    </ul>
  </div>
</div>
`;
